sdk/messaging/azwebpubsub/internal/policy_key_credential.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package internal
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/errorinfo"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type WebPubSubKeyCredentialPolicy struct {
```

Comment:
Nit: Can also remove `WebPubSub` prefix from this and accompanying types (I know it's internal but will shorten the names).